src/pip/_internal/commands/cache.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import absolute_import
+
+import logging
+import os
+import textwrap
+
+from pip._internal.cli.base_command import Command
+from pip._internal.exceptions import CommandError
+from pip._internal.utils.filesystem import find_files
+
+logger = logging.getLogger(__name__)
+
+
+class CacheCommand(Command):
+    """
+    Inspect and manage pip's caches.
+
+    Subcommands:
+        info:
+            Show information about the caches.
+        list [name]:
+            List filenames of packages stored in the cache.
+        remove <pattern>:
+            Remove one or more package from the cache.
+            `pattern` can be a glob expression or a package name.
+        purge:
+            Remove all items from the cache.
+    """
+    actions = ['info', 'list', 'remove', 'purge']
```

Comment:
Hmm, since this is another command with sub-commands (`pip config` is the other one), would it be OK with you to adapt this to that model for sub-command handling?

I imagine there would be two ways to go about it -- copy paste the code here, keeping the core logic for dispatching-to-subcommands the same (which we can refactor in a follow up PR) or move out the core logic for dispatching-to-subcommands into a dedicated class and then follow up with this PR being updated to do that.

I'm fine with either (with a preference for the latter, because it's cleaner) but I can definitely understand if you just want this PR to be just-add-the-cache command (i.e. the former) and we can look into refactoring to reduce the technical debt later. :)

Reply:
@pradyunsg I'm planning to poke at this again next week. I'll look into doing the second approach you mentioned when I do so. :+1: